src/linear_algebra/basic.lean

```diff
@@ -1042,6 +1044,22 @@ end
 @[reducible] def range_restrict (f : M →ₗ[R] M₂) : M →ₗ[R] f.range :=
 f.cod_restrict f.range f.mem_range_self
 
+section
+variables (R) (M)
+
+/-- Given an element `x` of a module `M` over `R`, the natural map from
+    `R` to scalar multiples of `x`.-/
+def span_singleton (x : M) : R →ₗ[R] M := linear_map.id.smul_right x
+
+/-- The range of `span_singleton x` is the span of `x`.-/
+lemma span_singleton_eq_range (x : M) : (span_singleton R M x).range = span R {x} :=
```

Comment:
The main difference is not the order (although maybe it would be a good idea to reverse my equation to match the format of the other.)  In the pre-existing lemma it's a `set` range, in the new lemma it's a `linear_map` range.

Reply:
Reversed the equation in cc0b73c; still curious about whether/how to unite with the similar lemma.